cockatrice/src/userconnection_information.cpp

```diff
@@ -0,0 +1,45 @@
+#include "userconnection_information.h"
+
+UserConnection_Information::UserConnection_Information(QString _serv, QString _port, QString _username, QString _password, bool _savePassword) :
+    server(_serv), port(_port), username(_username), password(_password), savePassword(_savePasword)
+{
+}
+
+void UserConnection_Information::addNewServer(QString _serv, QString _port, QString _username, QString _password, bool _savePassword)
+{
+    QSettings settings("Cockatrice", "server_details");
+
+    settings.beginGroup("server_details");
+    settings.beginWriteArray("server_details");
+    settings.setValue("server", server);
+    settings.setValue("port", port);
+    settings.setValue("username", username);
+    settings.setValue("password", password);
+    settings.setValue("savePassword", savePassword);
+    settings.endGroup();
+}
+
+QMap<QString, UserConnection_Information> UserConnection_Information::getServerInfo()
+{
+    QMap<QString, UserConnection_Information> tmp = new QMap<QString, UserConnection_Information>();
+    
+    int size = settings.beginReadArray("server_details");
+    for (int i = 0; i < size; i++)
+    {
+        settings.setArrayIndex(i);
+        
+        QString server_name = settings.value("server").toString();
```

Comment:
It seems like you're using server name as both the hostname and as the name of the profile - I think it would be better UX to let the users provide a name like

```
Server:
  Name: Woogerworks
  Host: cockatrice.woogerworks.com
  Post: 4747
  ...
```

Also this allows a very important thing - you can change the hostname of the server (for example using an ip address or an alternate dns name) without breaking the settings. Using hostname for both the key and the connection info means that changing the host will lose all associated settings